[OSS] feat(cli): add initial peering cli commands (#14423)
hashicorp/consul: website/content/api-docs/peering.mdx

- `Partition` `(string: "")` - <EnterpriseAlert inline /> The admin partition that the
  peering token is generated from. Uses `default` when not specified.

- `ServerExternalAddresses` `([]string: <optional>)` - A list of addresses to put

>   into the generated token. This could be used to specify load balancer(s) or
>   external IPs to reach the servers from the dialing side, and will override any server
>   addresses obtained from the "consul" service.

Author: I think it could be more clear what external IPs or load balancers we're talking about, and I'm not sure anymore that the last sentence adds value to why someone would need this.
```suggestion
into the generated token. You can specify one or more load balancers or external IPs that route external traffic to the cluster's Consul servers. 
```

>   addresses obtained from the "consul" service.

Reviewer: OK, as long as it's clear to users which cluster "this" refers to then I think this is fine.
